src/app/sessions/components/session-amendment-table/session-amendment-table.component.ts

```
@@ -0,0 +1,91 @@
+import { ChangeDetectionStrategy, Component, Input, OnChanges, ViewChild } from '@angular/core';
+import { MatPaginator, MatSort, MatTableDataSource } from '@angular/material';
+import * as moment from 'moment';
+import { SessionViewModel } from '../../models/session.viewmodel';
+
+@Component({
+  selector: 'app-session-amendment-table',
+  templateUrl: './session-amendment-table.component.html',
+  styleUrls: [],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SessionAmendmentTableComponent implements OnChanges {
+
+  @Input() sessions: SessionViewModel[];
+  @ViewChild(MatSort) sort: MatSort;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+
+  displayedColumns = [
+      'person',
+      'time',
+      'date',
+      'duration',
+      'room',
+      'sessionType',
+      'hearingParts',
+      'allocated',
+      'utilization',
+      'available',
+  ];
+
+  dataSource: MatTableDataSource<any>;
+  tableVisible;
+
+  constructor() {
+    this.dataSource = new MatTableDataSource(this.sessions);
+  }
+
+  parseTime(date: moment.Moment) {
+    return date.format('HH:mm');
+  }
+
+  humanizeDuration(duration) {
+      return moment.utc(moment.duration(duration).asMilliseconds()).format('HH:mm');
+  }
+
+  ngOnChanges() {
+      if (this.sessions) {
```

I would prefer to fail fast, so ```
if ( ! this.sessions ) {
  return;
}
```
and thn the rest